### PR TITLE
upstream sync for relu, transpose, fused_batch_norm op implementations

### PR DESCRIPTION
**Pull request by deven-amd**
This is in conjunction with the PRs I am filing to upstream ROCm support for these ops.

### CODE REVIEW

File: tensorflow/core/kernels/fused_batch_norm_op.cu.cc [hunk 1]
```diff
@@ -53,12 +53,7 @@ __global__ void InvVarianceToVarianceKernel(int nthreads, double epsilon,
                                             int sample_size, T* variance) {
   GPU_1D_KERNEL_LOOP(index, nthreads) {
     T inv_var = variance[index];
-#if GOOGLE_CUDA
     T var = __fdividef(1, inv_var * inv_var) - T(epsilon);
```
Comment (whchung) [inline tensorflow/core/kernels/fused_batch_norm_op.cu.cc]: @deven-amd do we have __fdividef now?

Comment (deven-amd) [reply tensorflow/core/kernels/fused_batch_norm_op.cu.cc]: yes!

I was surprised too

```
root@ixt-rack-04:/opt/rocm/include# grep fdividef -R .
./hip/hcc_detail/math_functions.h:float fdividef(float x, float y) { return x / y; }
./hip/hcc_detail/math_functions.h:float __fdividef(float x, float y) { return x / y; }
```

**Comment by deven-amd** [comment]
hmm...the build passes for me locally but fails CI, trying to isolate why

**Comment by deven-amd** [comment]
still unable to reproduce the failure locally...dont know what is going on.

will revert back the change in `relu_op_gpu.cu.cc` that disables the `Eigen::half` specialization for the `ReluGrad` template, since the compile error is within that code (which prior to this PR was disabled).

The compile errors are for references to the `half2` datatype (which are natively defined in the hip headers) and there are other files which contain half2 references and seem to compile fine.

**Comment by deven-amd** [comment]
posting the error here for future reference
```
tensorflow/core/kernels/relu_op_gpu.cu.cc:47:5: error: unknown type name 'half2'
    half2 gradient_h2 = reinterpret_cast<const half2*>(gradient)[index];
    ^
tensorflow/core/kernels/relu_op_gpu.cu.cc:47:48: error: unknown type name 'half2'
    half2 gradient_h2 = reinterpret_cast<const half2*>(gradient)[index];
                                               ^
tensorflow/core/kernels/relu_op_gpu.cu.cc:48:5: error: unknown type name 'half2'
    half2 feature_h2 = reinterpret_cast<const half2*>(feature)[index];
    ^
tensorflow/core/kernels/relu_op_gpu.cu.cc:48:47: error: unknown type name 'half2'
    half2 feature_h2 = reinterpret_cast<const half2*>(feature)[index];
                                              ^
tensorflow/core/kernels/relu_op_gpu.cu.cc:49:5: error: unknown type name 'half2'
    half2* p_backprop_h2 = reinterpret_cast<half2*>(backprop) + index;
    ^
tensorflow/core/kernels/relu_op_gpu.cu.cc:49:45: error: unknown type name 'half2'
    half2* p_backprop_h2 = reinterpret_cast<half2*>(backprop) + index;
                                            ^
tensorflow/core/kernels/relu_op_gpu.cu.cc:65:5: error: unknown type name 'half2'
    half2 backprop_h2 = __float22half2_rn(backprop_f2);
    ^
tensorflow/core/kernels/relu_op_gpu.cu.cc:65:25: error: use of undeclared identifier '__float22half2_rn'
    half2 backprop_h2 = __float22half2_rn(backprop_f2);
                        ^
8 errors generated.
```

**Pull request merged**